File: dotnet/SOLID/DeliveryOrder.cs
using dotnet.SOLID.Interface;
using dotnet.SOLID.Models;

namespace dotnet.SOLID;

public class DeliveryOrder : Order , IShippingCostCalculator
{
    private const decimal SHIPPING_COST = 10;

    public decimal CalculateShippingCost()
    {
        return Price + SHIPPING_COST;
    }
} 

File: dart_lang/lib/a_solid/c_liskov_substitution/b_post_refactor/delivery_order.dart
import 'package:dart_lang/a_solid/c_liskov_substitution/b_post_refactor/order.dart';
import 'package:dart_lang/a_solid/c_liskov_substitution/b_post_refactor/shipping_cost_calculator.dart';

class DeliveryOrder extends Order implements ShippingCostCalculator {
  static const double SHIPPING_COST = 10.0;

  @override
  double get totalPriceWithShippingCost => price ?? 0 + SHIPPING_COST;
}
